app/src/main/java/io/homeassistant/companion/android/settings/shortcuts/views/ManageShortcutsView.kt

```diff
@@ -0,0 +1,458 @@
+package io.homeassistant.companion.android.settings.shortcuts.views
+
+import android.os.Build
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material.Button
+import androidx.compose.material.Divider
+import androidx.compose.material.DropdownMenu
+import androidx.compose.material.DropdownMenuItem
+import androidx.compose.material.OutlinedButton
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.core.graphics.drawable.DrawableCompat
+import androidx.core.graphics.drawable.toBitmap
+import androidx.fragment.app.FragmentManager
+import com.maltaisn.icondialog.IconDialog
+import io.homeassistant.companion.android.common.R
+import io.homeassistant.companion.android.settings.shortcuts.ManageShortcutsSettingsFragment
+import io.homeassistant.companion.android.settings.shortcuts.ManageShortcutsViewModel
+
+@RequiresApi(Build.VERSION_CODES.N_MR1)
+@Composable
+fun ManageShortcutsView(
+    viewModel: ManageShortcutsViewModel,
+    iconDialog: IconDialog,
+    childFragment: FragmentManager
+) {
+    val context = LocalContext.current
+    var expandedEntity by remember { mutableStateOf(false) }
+    var expandedPinnedShortcuts by remember { mutableStateOf(false) }
+
+    LazyColumn(contentPadding = PaddingValues(20.dp)) {
+        item {
+            Text(
+                text = stringResource(id = R.string.shortcut_instruction_desc),
+                fontSize = 14.sp,
+                modifier = Modifier.padding(bottom = 10.dp)
+            )
+            Divider()
+        }
+
+        val shortcutCount = if (viewModel.canPinShortcuts)
+            ManageShortcutsSettingsFragment.MAX_SHORTCUTS + 1
+        else
+            ManageShortcutsSettingsFragment.MAX_SHORTCUTS
+
+        items(shortcutCount) { i ->

```

Comment:
There should be a function for creating the shortcut view itself rather than one giant function.